src/averageAges.js

```diff
@@ -37,7 +45,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(({ name, sex }) => withChildren
+    ? sex === 'f' && people.find(kid => kid.mother === name)
+    : sex === 'f');
+
+  const averAge = women.reduce((calc, woman) =>
+    (calc + (woman.died - woman.born)), 0) / women.length;
+
+  return averAge;
```

Comment:
Repeatable code move to helper function, `getAvarageAge` for example, and reuse it everywhere